src/pages/index.js

```diff
@@ -1,215 +1,275 @@
+const avatarInput = avatarModal.querySelector(".modal__input");
+const avatarSaveBtn = avatarModal.querySelector(".modal__save-button");
+
+//New Post Modal
+const newPostModal = page.querySelector("#new-post-modal");
+const newPostExitButton = page.querySelector(".modal__post-exit-button");
+const newPostButton = page.querySelector(".profile__add-button");
+const newPostUrl = page.querySelector("#link");
+const newPostCaption = page.querySelector("#caption");
+const buttonElement = page.querySelector("#new-post-btn");
+const newPostSaveBtn = newPostModal.querySelector(".modal__save-button");
+
+//Delete Modal
+const deleteModal = document.querySelector("#delete-card-modal");
+const deleteForm = document.forms["delete-card"];
+const deleteBtn = deleteModal.querySelector(".modal__delete-button");
+const cancelBtn = deleteModal.querySelector(".modal__cancel-button");
+
+//Preview Modal
+const previewModal = page.querySelector("#preview-modal");
+const previewExitButton = previewModal.querySelector(
+  ".modal__preview-exit-button"
+);
+const previewImage = previewModal.querySelector(".modal__preview-image");
+const previewDesc = previewModal.querySelector(".modal__preview-desc");
+
+//Universal function for close buttons
+// Find all close buttons
+const closeButtons = document.querySelectorAll(".modal__close");
+
+const myApi = new Api({
   baseUrl: "https://around-api.en.tripleten-services.com/v1",
   headers: {
-    authorization: "65d2d68a-31c1-4ecb-b4a7-2d6c51714b93",
+    authorization: "642fe59e-79a9-490f-b21e-64b2898ab938",
     "Content-Type": "application/json",
   },
 });
 
-api
-  .getAppInfo()
-  .then(([userData, cards]) => {
-    profileAvatar.src = userData.avatar;
-    profileNameElement.textContent = userData.name;
-    profileJobElement.textContent = userData.about;
-
-    cards.forEach((item) => {
-      const cardElement = getCardElement(item);
-      cardList.append(cardElement);
+closeButtons.forEach((button) => {
+  // Find the closest popup only once
+  const popup = button.closest(".modal");
+  // Set the listener
+  button.addEventListener("click", () => toggleModal(popup));
+});
+
+//Edit Profile Modal Functions/Handlers/Listeners
+function toggleModal(modal) {
+  modal.classList.toggle("modal_is-open");
+
+  if (modal.classList.contains("modal_is-open")) {
+    document.addEventListener("keyup", closeWithEscape);
+  } else {
+    document.removeEventListener("keyup", closeWithEscape);
+  }
+}
+
+const closeWithEscape = (evt) => {
+  if (evt.key === "Escape") {
+    const currModal = page.querySelector(".modal_is-open");
+    toggleModal(currModal);
+  }
+};
+
+function handleProfileFormSubmit(evt) {
+  function makeRequest() {
+    return myApi
+      .editUserInfo({ name: editModalName.value, about: editModalDesc.value })
+      .then((data) => {
+        profileName.textContent = data.name;
+        profileDesc.textContent = data.about;
+        toggleProfileModal();
+        evt.preventDefault();
```

Comment:
`evt.preventDefault();` is not needed here because `handleSubmit` contains it already

**Please, correct it everywhere in the code**